backend/entityservice/tasks/run.py

```diff
@@ -25,22 +26,26 @@ def prerun_check(project_id, run_id, parent_span=None):
             raise RunDeleted(run_id)
 
         try:
-            state = get_run_state_for_update(conn, run_id)
+            db_state = get_run_state_for_update(conn, run_id)
         except psycopg2.OperationalError:
             log.warning("Run started in another task. Skipping this race.")
             return
 
-        if state in {'running', 'completed', 'error'}:
+        if db_state in {'running', 'completed', 'error'}:
             log.warning("Run already started. Skipping")
             return
 
-        log.debug("Setting run as in progress")
+        # being very defensive here checking if the run state is already in the redis cache
+        if not is_run_missing(run_id):
```

Comment:
Small comment: shouldn't we first check the cache and then the db check? Like that if the check fail with the cache, we don't connect to the db which takes longer. But not important as it is used a single time per run.

Reply:
Fair point, I'll reorder.

I was actually wondering about using a redis lock for this whole section but didn't want to introduce it here.